### PR TITLE
Redesign: Publishing panel

### LINKED ISSUE #6514: Redesign: Update publishing panel
## Feature Description

See [Figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3483%3A1021)

- [ ] update titles
- [ ] update dropdowns
- [ ] update media selectors

## Additional Context
Screen Shot 2021-02-24 at 1 16 16 PM

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by miina**
## Relevant Technical Choices
- Uses the DropDownSelect from the design-system for Publish
- Uses `DropDownSelect` in the `AdvancedDropDown` to adjust the style for redesign. This will likely be changed once we have the "real" dropdown for fonts etc.

## To-do

- [ ] Test adjustments.

## User-facing changes
| Before | After |
|------|------|
| Screenshot 2021-03-15 at 16 26 47 | Screenshot 2021-03-15 at 16 25 40 |

## Testing Instructions
1. Open the Document Panel
2. Verify the Publish panel looks as in [Figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3483%3A1021)
3. Verify choosing time / author works as expected.
4. Now add a text element and verify that Selecting Font still works as expected.
5. 

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6514

### CODE REVIEW

File: assets/src/edit-story/components/form/advancedDropDown/dropDown.js [hunk 1]
```diff
@@ -211,17 +164,16 @@ function DropDown({
...
     <Container onKeyDown={handleKeyPress}>
       <DropDownSelect
-        onClick={toggleDropDown}
         aria-pressed={isOpen}
         aria-haspopup
         aria-expanded={isOpen}
         ref={ref}
-        lightMode={lightMode}
+        activeItemLabel={selectedOption?.name}
+        hasError={false}
```
Comment (barklund) [inline assets/src/edit-story/components/form/advancedDropDown/dropDown.js]: Why is this needed? Shouldn't `false` be the default anyway?

File: assets/src/design-system/components/mediaInput/index.js [hunk 2]
```diff
@@ -40,8 +40,8 @@ import { ReactComponent as Landscape } from './landscape.svg';
 import { MEDIA_VARIANTS } from './constants';
 
 const MediaRectangle = styled.section`
-  width: 64px;
-  height: 114px;
+  width: 100%;
+  height: 100%;
```
Comment (barklund) [inline assets/src/design-system/components/mediaInput/index.js:44]: Remember to update the sizing in Storybook - looks really weird now:

Screen Shot 2021-03-15 at 20 59 33

File: assets/src/edit-story/components/panels/design/link/link.js [hunk 3]
```diff
@@ -52,6 +52,11 @@ const IconText = styled(Text)`
   color: ${({ theme }) => theme.colors.fg.secondary};
 `;
 
+const MediaWrapper = styled.div`
+  width: 54px;
+  height: 54px;
+`;
```
Comment (barklund) [inline assets/src/edit-story/components/panels/design/link/link.js]: Could you not just do `const StyledMedia = styled(Media)``;` and let it accept either `...rest` or a `className`?

File: assets/src/edit-story/components/form/advancedDropDown/dropDown.js [hunk 4]
```diff
@@ -19,16 +19,14 @@
...
 import PropTypes from 'prop-types';
-import { rgba } from 'polished';
-import { __ } from '@web-stories-wp/i18n';
 
 /**
  * Internal dependencies
  */
-import { ReactComponent as DropDownIcon } from '../../../icons/dropdown.svg';
 import Popup from '../../popup';
+import DropDownSelect from '../../../../design-system/components/dropDown/select';
```
Comment (barklund) [inline assets/src/edit-story/components/form/advancedDropDown/dropDown.js:29]: Not completely sure if this change is related to this visual bug, but when you select an item and long-press the option, the layout shifts (see the dropdown height, it's probably due to the focus outline) until you release the mouse and the dropdown is closed.

It looks odd:

dropdown-focus-bug

File: assets/src/edit-story/components/panels/document/publish/author.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        />
+      ) : (
+        <AdvancedDropDown
+          options={queriedUsers}
+          primaryOptions={visibleOptions}
+          searchResultsLabel={__('Search results', 'web-stories')}
+          disabled={isSaving}
+          aria-label={__('Author', 'web-stories')}
+          {...dropDownParams}
+        />
```
Comment (barklund) [inline assets/src/edit-story/components/panels/document/publish/author.js]: Couldn't we collapse this into a single node? Just extract `isDisabled` and `placeholder` to `const`s above and we're more or less there, no?

File: assets/src/edit-story/components/panels/document/publish/publish.js [hunk 6]
```diff
@@ -204,63 +156,51 @@ function PublishPanel() {
...
-              onChange={handleChangePublisherLogo}
-              title={__('Select as publisher logo', 'web-stories')}
-              buttonInsertText={__('Select as publisher logo', 'web-stories')}
-              type={allowedImageMimeTypes}
-              ariaLabel={__('Publisher logo', 'web-stories')}
-              variant={MEDIA_VARIANTS.CIRCLE}
-            />
-          </MediaWrapper>
-        </Row>
         <HighlightRow isHighlighted={highlightPoster?.showEffect}>
```
Comment (barklund) [inline assets/src/edit-story/components/panels/document/publish/publish.js]: Should this perhaps now be:
```js
        <HighlightRow isHighlighted={highlightPoster?.showEffect || highlightLogo?.showEffect}>
```

Ping @embarks.

File: assets/src/edit-story/components/panels/document/publish/publish.js [hunk 7]
```diff
@@ -204,63 +156,51 @@ function PublishPanel() {
...
+                  ref={posterButtonRef}
+                  value={featuredMedia?.url}
+                  onChange={handleChangePoster}
+                  title={__('Select as poster image', 'web-stories')}
+                  buttonInsertText={__('Select as poster image', 'web-stories')}
+                  type={allowedImageMimeTypes}
+                  ariaLabel={__('Poster image', 'web-stories')}
+                />
+              </Sizer>
+            </MediaWrapper>
```
Comment (barklund) [inline assets/src/edit-story/components/panels/document/publish/publish.js]: The highlight rectangle around this one looks a bit off when coming from the checklist. I think it would be better if it was the actual bg element of the media, that had a blue outline (also following border-radius):

Screen Shot 2021-03-15 at 21 13 43

File: assets/src/edit-story/components/panels/document/publish/publish.js [hunk 8]
```diff
@@ -204,63 +156,51 @@ function PublishPanel() {
...
+                  type={allowedImageMimeTypes}
+                  ariaLabel={__('Publisher logo', 'web-stories')}
+                  variant={MEDIA_VARIANTS.CIRCLE}
+                />
+              </Sizer>
+            </MediaWrapper>
+            <LabelWrapper>
+              <Label>{__('Publisher logo', 'web-stories')}</Label>
+              <Required />
+            </LabelWrapper>
```
Comment (barklund) [inline assets/src/edit-story/components/panels/document/publish/publish.js:233]: Font-sizes for these appear wrong. Both are 16px but [should be](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=3398%3A94450) 14px and 12px respectively:

| Editor | Figma |
|-|-|
| Screen Shot 2021-03-15 at 21 18 21 | Screen Shot 2021-03-15 at 21 19 11 |

Also, what's the plan for the _"manage"_ link? Future ticket created?

**Review by barklund — COMMENTED** [review]
Looks great - just a few comments.

File: assets/src/edit-story/components/panels/document/publish/publish.js (hunk 6 above)
Comment (embarks) [reply assets/src/edit-story/components/panels/document/publish/publish.js]: @barklund that's right, since both inputs share the row now. this logic is just for the "flash" effect

File: assets/src/edit-story/components/form/advancedDropDown/dropDown.js (hunk 1 above)
Comment (miina) [reply assets/src/edit-story/components/form/advancedDropDown/dropDown.js]: It was complaining otherwise -- will check the design-system.

File: assets/src/edit-story/components/panels/document/publish/author.js (hunk 5 above)
Comment (miina) [reply assets/src/edit-story/components/panels/document/publish/author.js]: Didn't touch this part since it should be replaced anyway later with a "proper" dropdown, doesn't hurt to do it though.

File: assets/src/edit-story/components/panels/document/publish/publish.js (hunk 8 above)
Comment (miina) [reply assets/src/edit-story/components/panels/document/publish/publish.js:233]: > Font-sizes for these appear wrong.

Ok, will check with UX since supposedly if we need to change the font sizes of a preset, something might be off. 

On the Manage link: We had an issue somewhere for the Setting Page where the editor could choose between the Publisher logos instead of having just one set here. I think [this one](https://github.com/google/web-stories-wp/issues/5108).

File: assets/src/edit-story/components/form/advancedDropDown/dropDown.js (hunk 4 above)
Comment (miina) [reply assets/src/edit-story/components/form/advancedDropDown/dropDown.js:29]: There's no change to that part at all -- we still don't have the right dropdown component to use for this so currently, the PR is just switching out the button but not the list.

The same issue happens on `main`, too.

**Review by barklund — APPROVED** [review]
Looks great – and once again actually reducing overall line count!
